src/System.Linq/src/System/Linq/Reverse.cs

```diff
@@ -91,16 +91,7 @@ public override void Dispose()
             public TSource[] ToArray()
             {
                 TSource[] array = _source.ToArray();
-
-                // Array.Reverse() involves boxing for non-primitive value types, but
-                // checking that has its own cost, so just use this approach for all types.
-                for (int i = 0, j = array.Length - 1; i < j; ++i, --j)
-                {
-                    TSource temp = array[i];
-                    array[i] = array[j];
-                    array[j] = temp;
-                }
-
+                Array.Reverse<TSource>(array);
```

Comment:
I added type parameters on purpose to make sure the right overload always gets called... I mean, if the compiler picks up the generic overload anyway it doesn't matter after the fact, but just being defensive I guess.